pkg/sql/schemachanger/doc.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package schemachanger
+
+// The schemachanger package and its descendents contain the declarative schema
+// changer. This machinery is the latest iteration of how CockroachDB performs
+// schema changes, in contrast with the so-called legacy schema changer. This
+// doc file provides a high-level overview of its architecture, the aim being to
+// provide a useful introduction to the code and the design decisions behind
+// it.
+//
+// # Similarities and differences with the legacy schema changer
+//
+// The main challenge that any schema changer in CockroachDB must solve is to
+// perform these correctly while maintaining the "online" property: tables must
+// remain accessible to concurrent queries while they undergo schema changes.
+// Online schema changes are enabled by upholding the 2-version invariant, which
+// necessarily implies that schema changes may require multiple transactions to
+// complete. Upholding correctness requires maintaining some atomicity and
+// isolation properties which in turn means that schema changes which fail must
+// be rolled back somehow. Schema changes may legitimately fail, typically when
+// adding a constraint on a nonempty table.
+//
+// In CockroachDB, both in the legacy and in the declarative schema changer, for
+// schema changes involving more than one transaction. this sequence of
+// subsequent transactions is driven by a job; of type SCHEMA_CHANGE or
+// TYPEDESC_SCHEMA_CHANGE for the legacy schema changer, NEW_SCHEMA_CHANGE for
+// the declarative schema changer. In both cases the schema change gets rolled
+// back when the job's Resume method encounters an error, which eventually
```

Comment:
"Resume method encounters an error, after which its OnFailOrCancel method takes over and eventually transitions the job to the "failed" terminal state". 
Do you think adding those extra words in between is better?